arduino-ide-extension/src/electron-browser/theia/core/electron-main-menu-factory.ts

```diff
@@ -149,4 +151,119 @@ export class ElectronMainMenuFactory extends TheiaElectronMainMenuFactory {
     }
     return [];
   }
+
+  // Copied from 1.25.0 Theia as is to customize the enablement of the menu items.
+  // Source: https://github.com/eclipse-theia/theia/blob/ca417a31e402bd35717d3314bf6254049d1dae44/packages/core/src/electron-browser/menu/electron-main-menu-factory.ts#L125-L220
+  // See https://github.com/arduino/arduino-ide/issues/1533
```

Comment:
Cool!
This means we can also remove the workaround with `PlaceholderMenuNode` we applied in [cloud-sketchbook-contributions.ts](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-contributions.ts), [interface-scale.ts](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/browser/contributions/interface-scale.ts), [sketch-control.ts](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/browser/contributions/sketch-control.ts), [user-fields.ts](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/browser/contributions/user-fields.ts) and [user-fields.ts](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/browser/contributions/user-fields.ts), am I right?

(I'm not suggesting to necessarily do it in this PR, I just want to keep it in mind)

Reply:
Yes. Thanks for noticing